### PR TITLE
Added diagnostics packages

### PR DESCRIPTION
**Pull request by ivanpauno**
Added all diagnostics packages.

Also added yaml patch, avoiding it of building tests. This avoid gtest include files of being installed.
Delete your target/include/gtest folder if you are working in an existing output dir.

### CODE REVIEW

File: do_everything.sh [hunk 1]
```diff
@@ -173,6 +173,9 @@ if [[ $skip -ne 1 ]] ; then
     # Patch uuid - Avoiding stdlib.h include
     apply_patch $my_loc/patches/uuid.patch
 
+    # Patch yaml - Avoid building tests
+    apply_patch $my_loc/patches/yaml.patch
```
Comment (meyerj) [inline do_everything.sh]: rename to `yaml-cpp.patch` to match the library name in [do_everything.sh:121](https://github.com/Intermodalics/ros_android/pull/54/files#diff-82e657dd37ba5e5e07be9d0de6b1dbb6R121) and [do_everything.sh:333](https://github.com/Intermodalics/ros_android/pull/54/files#diff-82e657dd37ba5e5e07be9d0de6b1dbb6R333)?

**Review by meyerj — COMMENTED** [review]
Fails to compile for me:
```cpp
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:3381:3: error: cannot delete expression of type 'const internal::scoped_ptr< ::testing::internal::GTestFlagSaver>'                           
  delete gtest_flag_saver_;                                                                                                                                                                                                            
  ^      ~~~~~~~~~~~~~~~~~                                                                                                                                                                                                             
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:3661:11: error: out-of-line definition of 'TestInfo' does not match any declaration in 'testing::TestInfo'                                   
TestInfo::TestInfo(const std::string& a_test_case_name,                                                                                                                                                                                
          ^~~~~~~~                                                                                                                                                                                                                     
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:3661:11: error: constructor for 'testing::TestInfo' must explicitly initialize the member 'location_' which does not have a default constructor
/opt/ros_android/target/include/gtest/gtest.h:759:26: note: member is declared here                                                                                                                                                    
  internal::CodeLocation location_;                                                                                                                                                                                                    
                         ^                                                                                                                                                                                                             
/opt/ros_android/target/include/gtest/internal/gtest-internal.h:504:8: note: 'testing::internal::CodeLocation' declared here                                                                                                           
struct CodeLocation {                                                                                                                                                                                                                  
       ^                                                                                                                                                                                                                               
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:3710:11: error: no matching constructor for initialization of 'testing::TestInfo'                                                            
      new TestInfo(test_case_name, name, type_param, value_param,                                                                                                                                                                      
          ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                      
/opt/ros_android/target/include/gtest/gtest.h:728:3: note: candidate constructor not viable: requires 7 arguments, but 6 were provided                                                                                                 
  TestInfo(const std::string& test_case_name,                                                                                                                                                                                          
  ^                                                                                                                                                                                                                                    
/opt/ros_android/target/include/gtest/gtest.h:772:35: note: candidate constructor not viable: requires 1 argument, but 6 were provided                                                                                                 
  GTEST_DISALLOW_COPY_AND_ASSIGN_(TestInfo);                                                                                                                                                                                           
                                  ^                                                                                                                                                                                                    
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:6467:3: error: use of undeclared identifier 'g_argvs'                                                                                        
  g_argvs.clear();                                                                                                                                                                                                                     
  ^                                                                                                                                                                                                                                    
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:6469:5: error: use of undeclared identifier 'g_argvs'                                                                                        
    g_argvs.push_back(StreamableToString(argv[i]));                                                                                                                                                                                    
    ^                                                                                                                                                                                                                                  
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:7702:13: error: static declaration of 'SplitString' follows non-static declaration                                                           
static void SplitString(const ::std::string& str, char delimiter,                                                                                                                                                                      
            ^                                                                                                                                                                                                                          
/opt/ros_android/target/include/gtest/internal/gtest-internal.h:612:6: note: previous declaration is here                                                                                                                              
void SplitString(const ::std::string& str, char delimiter,                                                                                                                                                                             
     ^                                                                                                                                                                                                                                 
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:9008:13: error: functions that differ only in their return type cannot be overloaded                                                         
const char* StringFromGTestEnv(const char* flag, const char* default_value) {                                                                                                                                                          
      ~~~~~ ^                                                                                                                                                                                                                          
/opt/ros_android/target/include/gtest/internal/gtest-port.h:2549:13: note: previous declaration is here                                                                                                                                
std::string StringFromGTestEnv(const char* flag, const char* default_val);                                                                                                                                                             
~~~~~~~~~~~ ^                                                                                                                                                                                                                          
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:9553:31: error: use of undeclared identifier 'defined_test_names_'                                                                           
    for (DefinedTestIter it = defined_test_names_.begin();                                                                                                                                                                             
                              ^                                                                                                                                                                                                        
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:9554:16: error: use of undeclared identifier 'defined_test_names_'                                                                           
         it != defined_test_names_.end();                                                                                                                                                                                              
               ^                                                                                                                                                                                                                       
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:9570:29: error: use of undeclared identifier 'defined_test_names_'                                                                           
  for (DefinedTestIter it = defined_test_names_.begin();                                                                                                                                                                               
                            ^                                                                                                                                                                                                          
/opt/ros_android/catkin_ws/src/diagnostics/diagnostic_aggregator/gtest-1.7.0/gtest-all.cc:9571:14: error: use of undeclared identifier 'defined_test_names_'                                                                           
       it != defined_test_names_.end();                                                                                                                                                                                                
             ^                                                                                                                                                                                                                         
12 errors generated.                                                                                                                                                                                                                   
CMakeFiles/analyzer_loader.dir/build.make:86: recipe for target 'CMakeFiles/analyzer_loader.dir/gtest-1.7.0/gtest-all.cc.o' failed                                                                                                     
make[2]: *** [CMakeFiles/analyzer_loader.dir/gtest-1.7.0/gtest-all.cc.o] Error 1                                                                                                                                                       
make[2]: *** Waiting for unfinished jobs....                                                                                                                                                                                           
CMakeFiles/Makefile2:328: recipe for target 'CMakeFiles/analyzer_loader.dir/all' failed                                                                                                                                                
make[1]: *** [CMakeFiles/analyzer_loader.dir/all] Error 2                                                                                                                                                                              
make[1]: *** Waiting for unfinished jobs....                                                          
```

The test is compiled unconditionally and does not respect `CATKIN_ENABLE_TESTING`.

**Comment by meyerj** [comment]
> Fails to compile for me:

Fixed after having cleaned the gtest artifacts in the target-space. I should have read the description.

File: do_everything.sh (hunk 1 above)
Comment (ivanpauno) [reply do_everything.sh]: Sounds good, already modified.

**Comment by ivanpauno** [comment]
> > Fails to compile for me:
> 
> Fixed after having cleaned the gtest artifacts in the target-space. I should have read the description.

Yes the problem was that yaml-cpp installed gtest includes when building tests, and analyzer_loader was  built based on a different version of gtest.
I didn't drop analyzer_loader, maybe it can been also dropped.

**Review by jubeira — APPROVED** [review]
Tested on my side as well; I think this one can be merged after fixing the conflicts.

**Pull request merged**